src/registrar/models/models.py

```diff
@@ -0,0 +1,422 @@
+    agency = models.ManyToManyField(FederalAgency)
+
+
+
+class AuthorizingOfficial(TimeStampedModel, ContactModel, AddressModel):
+    # contact's name
+    name = models.CharField(max_length=2)
+    # alternative contacting instructions (could also be foreign key to AO record)
+    alt_contact_info = models.CharField(max_length=2)
+    # an AO may or may not have a user in the system
+    user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True)
+
+
+class Organization(TimeStampedModel, AddressModel):
+    name = models.CharField(max_length=2)
+    entity = models.ForeignKey(GovEntity, on_delete=models.CASCADE)
+    authorizing_official = models.ForeignKey(AuthorizingOfficial, on_delete=models.CASCADE)
+
+
+class Rejection(models.Model):
+    reason = models.CharField(max_length=2)
+
+
+class Application(TimeStampedModel):
+    # user who is making the request
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    # the person who says this user is allowed to request this domain
+    authorizing_official = models.ForeignKey(AuthorizingOfficial, on_delete=models.CASCADE)
+    # the government who CISA says is allowed to request a .gov domain
+    authorizing_entity = models.ForeignKey(GovEntity, on_delete=models.CASCADE)
+    # the subdivision or department this user belongs to
+    organization = models.ForeignKey(Organization, on_delete=models.PROTECT)
+    domain_name = models.CharField(max_length=2)
+    is_exception_requested = models.BooleanField()
+    # if the user wants an ineligible domain name, they may give a reason for their request
+    exception_rationale = models.CharField(max_length=2)
+    is_expedited_requested = models.BooleanField()
+    # if the user wants a fast response, they may give a reason for their request
+    expedited_rationale = models.CharField(max_length=2)
+    hsts_acknowledged = models.BooleanField()
+    # reason for requesting the domain
+    purpose = models.CharField(max_length=2)
+    # Choices ??? -- also, TODO: use a state machine to model the business logic
+    status = models.CharField(max_length=2)
+    # choices ???
+    rejection_codes = models.ManyToManyField(Rejection)
+    assignee = models.ForeignKey(User, null=True, on_delete=models.SET_NULL, related_name="assigned")
+
+
+class ApplicationStatusHistory(TimeStampedModel):
+    application = models.ForeignKey(Application, on_delete=models.CASCADE)
+    prior_status = models.CharField(max_length=2)
+    status = models.CharField(max_length=2)
+    change_note = models.CharField(max_length=2)
+
+
+class Domain(TimeStampedModel, CommonEPPModel):
+    # fully qualified name (unique)
+    name  = models.CharField(max_length=2)
+    # identifier assigned to the domain object when created
+    roid  = models.CharField(max_length=2) # NS1_EXAMPLE1-REP
+    # the date and time of the domain object's registration expiration
+    ex_date = models.CharField(max_length=2)
```

Comment:
Can we use a more legible field name like `expiration_date`? Do we want a `DateTimeField` for this?